### PR TITLE
Circulation community pool bonded supply

### PR DESCRIPTION
**Pull request by egaxhaj-figure**
## Description

Implement Token statistics:
* Circulation supply
* Community Pool
* Bonded supply

closes: #188 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/explorer/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit and integration
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: service/src/main/kotlin/io/provenance/explorer/grpc/v1/BankGrpcClient.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        bankClient.supplyOf(BankOuterClass.QuerySupplyOfRequest.newBuilder().setDenom(denom).build()).amount.amount
+
+    fun getCommunityPoolAmount(denom: String): String =
+        distClient.communityPool(DistOuterClass.QueryCommunityPoolRequest.newBuilder().build()).poolList
+            .filter { it.denom == denom }[0]?.amount!!.toDecCoin()
+
+    fun getBondedAmount(): String =
+        stakingClient.validators(
+            StakingOuterClass.QueryValidatorsRequest.newBuilder()
+                .setStatus(Staking.BondStatus.BOND_STATUS_BONDED.toString())
```
Comment (crussell-figure) [inline service/src/main/kotlin/io/provenance/explorer/grpc/v1/BankGrpcClient.kt:58]: No need to filter on BONDED only. Even an UNBONDED validator still has a token value.

File: service/src/main/kotlin/io/provenance/explorer/service/TokenSupplyService.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+        val currentSupply = bankClient.getCurrentSupply(NHASH).toBigDecimal()
+        val communityPoolSupply = bankClient.getCommunityPoolAmount(NHASH).toBigDecimal()
+        val bondedSupply = bankClient.getBondedAmount().toBigDecimal()
+
+        val circulation = (currentSupply - communityPoolSupply - bondedSupply - markerAmount).toCoinStr(NHASH)
+        val communityPool = communityPoolSupply.toCoinStr(NHASH)
+        val bonded = bondedSupply.toCoinStr(NHASH)
+
+        TokenSupply(circulation, communityPool, bonded)
```
Comment (crussell-figure) [inline service/src/main/kotlin/io/provenance/explorer/service/TokenSupplyService.kt:30]: Need to have `total_supply` as well, which is just `current_supply`

**Review by crussell-figure — CHANGES_REQUESTED** [review]
Just a couple minor things.

File: service/src/main/kotlin/io/provenance/explorer/service/TokenSupplyService.kt (hunk 2 above)
Comment (egaxhaj-figure) [reply service/src/main/kotlin/io/provenance/explorer/service/TokenSupplyService.kt]: Added `current_supply`

File: service/src/main/kotlin/io/provenance/explorer/grpc/v1/BankGrpcClient.kt (hunk 1 above)
Comment (egaxhaj-figure) [reply service/src/main/kotlin/io/provenance/explorer/grpc/v1/BankGrpcClient.kt:58]: Included `unbonded` in calculation and also added to the response as separate values.

```
{
    "currentSupply": {
        "amount": "100000000000000000000",
        "denom": "nhash"
    },
    "circulation": {
        "amount": "8302147516414584507.978653519831055111",
        "denom": "nhash"
    },
    "communityPool": {
        "amount": "14835476107678.021346480168944889",
        "denom": "nhash"
    },
    "bonded": {
        "amount": "6806714289481807801",
        "denom": "nhash"
    },
    "unBonded": {
        "amount": "4142389949975900014",
        "denom": "nhash"
    }
}
```

Comment (crussell-figure) [reply service/src/main/kotlin/io/provenance/explorer/grpc/v1/BankGrpcClient.kt:58]: Per Ira, "bonded" tokens are all token counts across all validators. No need to have separate values for that. If you remove the status setting, itll work as intended across all validators.

**Review by crussell-figure — APPROVED** [review]

**Pull request merged**